### PR TITLE
Add admin functionality to histories API

### PR DESCRIPTION
**Pull request by thomcuddihy**
Added the ability for Admins to query all histories using the API.

This was implemented as part of a new tool for managing users for Galaxy Australia, and generating stats/graphs for reporting measures.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/histories.py [hunk 1]
```diff
@@ -144,8 +146,12 @@ def index(self, trans, deleted='False', **kwd):
         filters = []
         # support the old default of not-returning/filtering-out deleted histories
         filters += self._get_deleted_filter(deleted, filter_params)
-        # users are limited to requesting only their own histories (here)
-        filters += [self.app.model.History.user == current_user]
+        # check if user is admin and optional parameter 'all' is True
+        admin = trans.user_is_admin
+        all_histories = util.string_as_bool(kwd.get('all', False))
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/histories.py:150]: Can you throw ``galaxy.exceptions.AdminRequiredException`` if all_histories is true but admin is not? I think the new if would read a bit better too if you did that.

File: lib/galaxy/webapps/galaxy/api/histories.py [hunk 2]
```diff
@@ -144,8 +146,17 @@ def index(self, trans, deleted='False', **kwd):
         filters = []
         # support the old default of not-returning/filtering-out deleted histories
         filters += self._get_deleted_filter(deleted, filter_params)
-        # users are limited to requesting only their own histories (here)
-        filters += [self.app.model.History.user == current_user]
+        # check if user is admin and get optional parameter 'all' 
+        admin = trans.user_is_admin
```
Comment (jdavcs) [inline lib/galaxy/webapps/galaxy/api/histories.py]: Minor suggestion. You only need to check the user's admin status if `all_history` is `True`. So you could move this line inside the `if` stmt; but you could also get rid of it completely and do `if not trans.user_is_admin` instead - which, I think, will read even better.

File: lib/galaxy/webapps/galaxy/api/histories.py (hunk 1 above)
Comment (thomcuddihy) [reply lib/galaxy/webapps/galaxy/api/histories.py:150]: Added. Thanks for the exception tip.

File: lib/galaxy/webapps/galaxy/api/histories.py (hunk 2 above)
Comment (thomcuddihy) [reply lib/galaxy/webapps/galaxy/api/histories.py]: Added. Thanks mate

Comment (jdavcs) [reply lib/galaxy/webapps/galaxy/api/histories.py]: looks great!

**Comment by martenson** [comment]
@galaxybot test this

**Comment by martenson** [comment]
Thank you @thomcuddihy for the contribution!

**Pull request merged**